OneCommunityGlobal/HighestGoodNetworkApp
# Ilya consumables add page

# Description
<img width="754" alt="consumables-add" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/a75e41b6-91e3-491d-a36c-cf2aece3c21a">

Adds functionality for the user to navigate to /consumables/add, enter the required information (consumable name, description, measurement units, size) and create a new consumable type in the system.

The front end validates the input to have a name that is at least 4 characters long and a description that is at least 10 characters long. The back end checks if a consumable under the same name already exists.

## Related PRS (if any):
This frontend PR is related to the [back end PR#737](https://github.com/OneCommunityGlobal/HGNRest/pull/737)
To test this front end PR you need to checkout the #737 back end PR, branch **ilya_consumables_add_page**
…

## Main changes explained:
- /consumables/add route added to the front end router
- AddConsumables component includes a form with validation that makes a POST request to the backend
- The new component is styled to accommodate smaller screens 
- Necessary Redux actions and Reducers created
- Existing files formatted with ESLint and Prettier

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as any user
5. go to [http://localhost:3000/bmdashboard ](http://localhost:3000/bmdashboard) and log in again
6. go to [http://localhost:3000/bmdashboard/consumables/add](http://localhost:3000/bmdashboard/consumables/add), enter some dummy data, press "Add Consumable" when the button is enabled
7. A success message should appear if everything is acceptable or a warning message if not (such as a duplicate value)
<img width="1127" alt="consumables-fe" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/0f6334ab-6c81-49cc-82bf-3a99fbd787a2">
8. In MongoDB Compass open hgnData_dev.buildingInventoryTypes collection. Press "Find", scroll all the way to the bottom or search by name and verify that the entered information is saved in the database
<img width="1419" alt="compass" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/08034d00-5c9e-4aeb-8232-f7d6037cd935">


## Screenshots or videos of changes:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105448137/a5600668-ccf6-4e3b-b5e7-b52be3e7cb7f


## Note:
Examples of construction consumables can include nails, screws, batteries, lumber, etc.
